### DISCUSSION TITLE
How to have multiple queries in NodeJS

### DISCUSSION THREAD

**Discussion by luandro** [vercel/vercel#4096] [General] [answered]
In NodeJS, I understand that by creating a `[name].js` file for example, will give me a `name` query.
But my api requires multiple queries, how could I handle that? For example `name` and `lastname`. I would image having a file named: `[name][lastname].js`. But that doesn't work.

Any tips? :thinking:

**Comment by mcsdevv** [comment]
Hey! For that you can use query strings, it's not something that is supported by path segments currently.

**Reply by luandro** [reply]
Thanks @msweeneydev!

It's a bit of an ugly hack, but might do... Really no other option?

**Comment by styfle** [comment, marked as answer]
Try creating a file `/api/[firstname]/[lastname].js` to create multiple [path segments](https://vercel.com/docs/v2/serverless-functions/introduction#path-segments).

Another alternative is to create a file without path segments, such as `/api/handle-name.js`, and then add a `now.json` file with [rewrites](
https://vercel.com/docs/configuration#project/rewrites) like the following:

```json
{
  "rewrites": [
    { "source": "/v1/:first/:last", "destination": "/api/handle-name" }
  ]
}
```

**Comment by FernandoKGA** [comment]
@styfle It's possible to use the same structure but passing from path segments, but using `routes`, to query?

Example:

```json
{
    "routes": [
        { "src": "/orders/(?<order>someregexhere)/update", "methods": ["POST"], "dest": "/api/orders/[order]/update.js?order=$order" }
    ]
}
```

The example above don't redirect the `order` to the query string. Maybe I'm mixing two concepts that can't be mixed. What would be the correct way to do so? I do want to use the path segment, but I also need the query parameters...

This one also don't insert into `req.query`.
```json
{
    "routes": [
        { "src": "/orders/(?<order>someregexhere)/update", "methods": ["POST"], "dest": "/api/orders/[order]/update.js" }
    ]
}
```

**Reply by FernandoKGA** [reply]
Hmm, well... It does work, but only if I do something this:

```json
{
    "routes": [
        { "src": "/orders/(?<order>someregexhere)/update", "methods": ["POST"], "dest": "/api/orders/[order]/update?order=$order" }
    ]
}
```

**Reply by styfle** [reply]
Hi @FernandoKGA 

`routes` acts differently than `rewrites`. If you choose `routes`, then there is no additional syntax sugar. What you write is what you get. So if you need a query string in the `destination`, then you'll need to write it out like that.
